src/cata_utility.h

```diff
@@ -10,6 +10,34 @@ class Creature;
 class map_item_stack;
 struct tripoint;
 
+/**
+* A class template that mimics a 2 dimensional array.
+*/
+template<typename T>
+class array_2d {
+private:
+    size_t size_x, size_y;
+    std::vector<T> _array;
+public:
+    array_2d( size_t x, size_t y ) : size_x( x ), size_y( y ), _array( std::vector<T>( x * y ) ) // Make sure capacity is sufficient
```

Comment:
The initialization of `_array` is a bit strange. A simple `_array( x * y )` should do.